packages/plugins/typescript/vue-urql/src/visitor.ts

```diff
@@ -98,8 +98,8 @@ export function use${operationName}<R = ${operationResultType}>(options: Omit<Ur
     }
 
     return `
-export function use${operationName}(options: Omit<Urql.Use${operationType}Args<never, ${operationVariablesTypes}>, 'query'> = {}) {
```

Comment:
Even leaving the default in place works for me in all cases, no matter if I
- have required variables,
- have only optional variables,
- have no variables at all.

With ` @urql/vue ^1.1.1`, the only required change is your change in line 102.

I would prefer to leave the default (line 101) in place, to make it a non-breaking change for current users.

Reply:
I use typescript version 5.

Reply:
Can we merge and publish these changes?

Reply:
This PR is stale for more than a month now, can we get this merged or provide reasons for not merging it?

Reply:
Variables & variables fiels maybe ref :
```patch
-export function use${operationName}(options: Omit<Urql.Use${operationType}Args<never, ${operationVariablesTypes}>, 'query'> = {}) {
-  return Urql.use${operationType}<${operationResultType}>({ query: ${documentVariableName}, ...options });
+type MaybeRef<T> = T | Ref<T>; 
+export function use${operationName}(options: Omit<Urql.Use${operationType}Args<never, MaybeRef<{[K in keyof ${operationVariablesTypes}]: MaybeRef<${operationVariablesTypes}[K]>}>>, 'query'>) {
+  return Urql.use${operationType}<${operationResultType}, ${operationVariablesTypes}>({ query: ${documentVariableName}, ...options });

```